src/api/Bolt.tsx

```diff
@@ -0,0 +1,243 @@
+import {
+    InputBindingConstructor,
+    NumberBinding,
+} from './controllers';
+
+import React from 'react';
+import { render } from 'react-dom';
+import BoltGUI from '../components/BoltGUI';
+import { ProviderWrapperComponent } from '../components/ProviderWrapperComponent';
+import { ComponentPair, PanelData, WindowData } from '../types';
+import store from '../redux/store';
+import {
+    addPanel,
+    clearPanel,
+    setActivePanel,
+    addWindow,
+    addComponent,
+    removeComponent,
+    updateComponent,
+} from '../redux';
+import { defaultAttributes } from '../components/Themeable';
+import { ButtonInputComponent, Label } from '..';
+
+export class BoltClass
+{
+    protected static _bindingTests: InputBindingConstructor[];
+    protected _defaultWindow: WindowData;
+
+    constructor(options: Partial<WindowData> = {})
+    {
+        const data = { ...defaultAttributes.window, ...options };
+
+        this._defaultWindow = this.createWindow(data);
+
+        BoltClass._bindingTests = [];
+        this.addTest(NumberBinding);
+    }
+
+    public init(): void
+    {
+        let element = document.getElementById('editor-holder');
+
+        if (!element)
+        {
+            element = document.createElement('div');
+            element.id = 'editor-holder';
+            document.body.appendChild(element);
+        }
+
+        render(
+            <ProviderWrapperComponent>
+                <BoltGUI/>
+            </ProviderWrapperComponent>
+            , element);
+    }
+
+    /**
+     * Creates a new panel and adds it to a window
+     * @param window - window to add the panel too
+     * This will default to the default window created by Bolt
+     */
+    public createPanel(options?: PanelData, windowID: string = this._defaultWindow.id): PanelData
+    {
+        store.dispatch(addPanel(options, windowID));
+
+        return options;
+    }
+
+    /**
+     * Creates a new panel and adds it to a window
+     * @param window - window to add the panel too
+     * This will default to the default window created by Bolt
+     */
+    public clearPanel(id: string, destroyChildren: boolean): void
+    {
+        store.dispatch(clearPanel(id, destroyChildren));
+    }
+
+    /**
+     * Creates a new panel and adds it to a window
+     * @param window - window to add the panel too
+     * This will default to the default window created by Bolt
+     */
+    public setActivePanel(id: string): void
+    {
+        store.dispatch(setActivePanel(id));
+    }
+
+    // TODO: implement an addPanel() to allow for a panel to be added back after being removed
+
+    /**
+     * Removes a panel from a window
+     * @param panel - panel to remove
+     */
+    // public removePanel(panel: PanelController): void
```

Comment:
I thought you added these functions already

Reply:
it is not implemented yet, and will not be in the scope of this PR, however it was causing issues with the react compiler that were preventing me from building/running the project. Commented out until implementation is completed.